In [2]:
import pandas as pd

In [3]:
data = pd.read_csv(r'C:\Users\MY PC\Documents\GitHub\CG-DA-Python\Datasets\FoodPrice_in_Turkey.csv', encoding = 'ISO-8859-1')

In [4]:
data.head()

Place  ProductId    ProductName  UmId UmName  Month  Year  \
0  National Average         52  Rice - Retail     5     KG      5  2013   
1  National Average         52  Rice - Retail     5     KG      6  2013   
2  National Average         52  Rice - Retail     5     KG     11  2013   
3  National Average         52  Rice - Retail     5     KG     12  2013   
4  National Average         52  Rice - Retail     5     KG      1  2014   

    Price  
0  4.4920  
1  4.5786  
2  4.7865  
3  5.1337  
4  5.5099

In [5]:
data

Place  ProductId              ProductName  UmId UmName  \
0     National Average         52            Rice - Retail     5     KG   
1     National Average         52            Rice - Retail     5     KG   
2     National Average         52            Rice - Retail     5     KG   
3     National Average         52            Rice - Retail     5     KG   
4     National Average         52            Rice - Retail     5     KG   
...                ...        ...                      ...   ...    ...   
7376             Izmir        502  Cocoa (powder) - Retail     5     KG   
7377             Izmir        502  Cocoa (powder) - Retail     5     KG   
7378             Izmir        502  Cocoa (powder) - Retail     5     KG   
7379             Izmir        502  Cocoa (powder) - Retail     5     KG   
7380             Izmir        502  Cocoa (powder) - Retail     5     KG   

      Month  Year    Price  
0         5  2013   4.4920  
1         6  2013   4.5786  
2        11  2013   4.7865  
3        12  2013   5.1337  
4         1  2014   5.5099  
...     ...   ...      ...  
7376      8  2019  47.9040  
7377      9  2019  49.1176  
7378     10  2019  50.8347  
7379     11  2019  51.7985  
7380     12  2019  51.6071  

[7381 rows x 8 columns]

In [6]:
products_in_each_place = data.groupby('Place')['ProductName'].unique()

In [7]:
products_in_each_place

Place
Ankara              [Rice - Retail, Wheat flour - Retail, Beans (w...
Istanbul            [Rice - Retail, Wheat flour - Retail, Beans (w...
Izmir               [Rice - Retail, Wheat flour - Retail, Beans (w...
National Average    [Rice - Retail, Wheat flour - Retail, Beans (w...
Name: ProductName, dtype: object

In [8]:
data['ProductName'].unique().sort()

In [9]:
products = data['ProductName'].unique()
products.sort()
products

array(['Apples (red) - Retail', 'Apples - Retail', 'Bananas - Retail',
       'Beans (white) - Retail', 'Bread (common) - Retail',
       'Bread (pita) - Retail', 'Bulgur - Retail', 'Cabbage - Retail',
       'Cauliflower - Retail', 'Cheese - Retail', 'Chickpeas - Retail',
       'Cocoa (powder) - Retail', 'Coffee (instant) - Retail',
       'Coffee - Retail', 'Cucumbers (greenhouse) - Retail',
       'Cucumbers - Retail', 'Eggplants - Retail', 'Eggs - Retail',
       'Electricity - Retail', 'Fish (fresh) - Retail',
       'Fuel (gas) - Retail', 'Fuel (petrol-gasoline) - Retail',
       'Garlic - Retail', 'Groundnuts (shelled) - Retail',
       'Lentils - Retail', 'Meat (chicken) - Retail',
       'Meat (mutton) - Retail', 'Meat (veal) - Retail',
       'Milk (pasteurized) - Retail',
       'Milk (powder, infant formula) - Retail', 'Oil (olive) - Retail',
       'Oil (sunflower) - Retail', 'Onions - Retail', 'Oranges - Retail',
       'Pasta - Retail', 'Peas (green, dry) - Retail',
   

In [36]:
type(products)

numpy.ndarray

In [10]:
places = data['Place'].unique()
places.sort()
print(places)

['Ankara' 'Istanbul' 'Izmir' 'National Average']


In [11]:
places

array(['Ankara', 'Istanbul', 'Izmir', 'National Average'], dtype=object)

In [12]:
display(places)

array(['Ankara', 'Istanbul', 'Izmir', 'National Average'], dtype=object)

In [13]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7381 entries, 0 to 7380
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Place        7381 non-null   object 
 1   ProductId    7381 non-null   int64  
 2   ProductName  7381 non-null   object 
 3   UmId         7381 non-null   int64  
 4   UmName       7381 non-null   object 
 5   Month        7381 non-null   int64  
 6   Year         7381 non-null   int64  
 7   Price        7381 non-null   float64
dtypes: float64(1), int64(4), object(3)
memory usage: 461.4+ KB


# Bộ dữ liệu có 7381 dòng và 8 cột

# Place, ProductId, ProductName, UmId, UmName, Month và Year là những thuộc tính thuộc thang đo Nominal (Định danh)
# Còn Price là thuộc tính thuộc thang đo Ratio (Tỉ lệ)

In [40]:
average_price = data.groupby(['ProductId', 'ProductName'])['Price'].mean()

In [41]:
type(average_price)

pandas.core.series.Series

In [45]:
average_price

ProductId  ProductName                                           
52         Rice - Retail                                               7.950325
58         Wheat flour - Retail                                        3.253174
66         Beans (white) - Retail                                      9.572830
83         Potatoes - Retail                                           2.297231
92         Eggs - Retail                                               0.505925
94         Meat (chicken) - Retail                                     9.683471
97         Sugar - Retail                                              4.482942
105        Garlic - Retail                                            16.610370
108        Lentils - Retail                                            6.651817
112        Pasta - Retail                                              3.496637
114        Tomatoes - Retail                                           3.393658
119        Groundnuts (shelled) - Retail              

In [46]:
average_price.sort_values()

ProductId  ProductName                                           
92         Eggs - Retail                                               0.505925
491        Electricity - Retail                                        0.513580
181        Cabbage - Retail                                            1.723395
185        Salt - Retail                                               2.048213
173        Onions - Retail                                             2.141539
233        Cucumbers (greenhouse) - Retail                             2.206807
83         Potatoes - Retail                                           2.297231
229        Apples (red) - Retail                                       2.392193
384        Bread (pita) - Retail                                       2.404545
304        Transport (public) - Retail                                 2.468031
212        Peas (green, dry) - Retail                                  2.900029
360        Oranges - Retail                           

In [42]:
average_price_df = pd.DataFrame(data.groupby(['ProductId', 'ProductName'])['Price'].mean())

In [43]:
type(average_price_df)

pandas.core.frame.DataFrame

In [35]:
average_price

Price
ProductId ProductName                                                   
52        Rice - Retail                                         7.950325
58        Wheat flour - Retail                                  3.253174
66        Beans (white) - Retail                                9.572830
83        Potatoes - Retail                                     2.297231
92        Eggs - Retail                                         0.505925
94        Meat (chicken) - Retail                               9.683471
97        Sugar - Retail                                        4.482942
105       Garlic - Retail                                      16.610370
108       Lentils - Retail                                      6.651817
112       Pasta - Retail                                        3.496637
114       Tomatoes - Retail                                     3.393658
119       Groundnuts (shelled) - Retail                        21.832122
124       Bread (common) - Retail                               4.463293
151       Coffee - Retail                                      42.468075
173       Onions - Retail                                       2.141539
176       Tea (green) - Retail                                  4.137000
181       Cabbage - Retail                                      1.723395
183       Meat (mutton) - Retail                               45.557209
184       Oil (sunflower) - Retail                              8.755174
185       Salt - Retail                                         2.048213
209       Coffee (instant) - Retail                            43.295493
212       Peas (green, dry) - Retail                            2.900029
229       Apples (red) - Retail                                 2.392193
231       Cauliflower - Retail                                  3.274868
233       Cucumbers (greenhouse) - Retail                       2.206807
239       Oil (olive) - Retail                                 27.274547
244       Chickpeas - Retail                                   10.347860
254       Bananas - Retail                                      8.224784
258       Bulgur - Retail                                       3.516712
281       Tea - Retail                                         26.224393
285       Fuel (petrol-gasoline) - Retail                       8.592519
304       Transport (public) - Retail                           2.468031
317       Tomatoes (paste) - Retail                             8.085128
341       Fuel (gas) - Retail                                  83.976703
345       Apples - Retail                                       3.885039
360       Oranges - Retail                                      3.124280
364       Yogurt - Retail                                       5.364898
384       Bread (pita) - Retail                                 2.404545
388       Cucumbers - Retail                                    3.411326
401       Fish (fresh) - Retail                                22.698257
404       Spinach - Retail                                      3.240587
414       Cheese - Retail                                      25.522875
433       Meat (veal) - Retail                                 41.270904
434       Eggplants - Retail                                    3.882645
442       Milk (powder, infant formula) - Retail              108.287696
450       Zucchini - Retail                                     3.837423
463       Milk (pasteurized) - Retail                           3.679687
465       Wage (non-qualified labour, non-agricultural) -...   40.548182
491       Electricity - Retail                                  0.513580
492       Tea (herbal) - Retail                                 5.434838
493       Water - Retail                                        5.384253
502       Cocoa (powder) - Retail                              43.802339

In [30]:
type(average_price)

pandas.core.frame.DataFrame

In [31]:
type(data)

pandas.core.frame.DataFrame

In [50]:
average_price.sort_values()

ProductId  ProductName                                           
92         Eggs - Retail                                               0.505925
491        Electricity - Retail                                        0.513580
181        Cabbage - Retail                                            1.723395
185        Salt - Retail                                               2.048213
173        Onions - Retail                                             2.141539
233        Cucumbers (greenhouse) - Retail                             2.206807
83         Potatoes - Retail                                           2.297231
229        Apples (red) - Retail                                       2.392193
384        Bread (pita) - Retail                                       2.404545
304        Transport (public) - Retail                                 2.468031
212        Peas (green, dry) - Retail                                  2.900029
360        Oranges - Retail                           

In [52]:
average_price_df.sort_values('ProductName')

Price
ProductId ProductName                                                   
229       Apples (red) - Retail                                 2.392193
345       Apples - Retail                                       3.885039
254       Bananas - Retail                                      8.224784
66        Beans (white) - Retail                                9.572830
124       Bread (common) - Retail                               4.463293
384       Bread (pita) - Retail                                 2.404545
258       Bulgur - Retail                                       3.516712
181       Cabbage - Retail                                      1.723395
231       Cauliflower - Retail                                  3.274868
414       Cheese - Retail                                      25.522875
244       Chickpeas - Retail                                   10.347860
502       Cocoa (powder) - Retail                              43.802339
209       Coffee (instant) - Retail                            43.295493
151       Coffee - Retail                                      42.468075
233       Cucumbers (greenhouse) - Retail                       2.206807
388       Cucumbers - Retail                                    3.411326
434       Eggplants - Retail                                    3.882645
92        Eggs - Retail                                         0.505925
491       Electricity - Retail                                  0.513580
401       Fish (fresh) - Retail                                22.698257
341       Fuel (gas) - Retail                                  83.976703
285       Fuel (petrol-gasoline) - Retail                       8.592519
105       Garlic - Retail                                      16.610370
119       Groundnuts (shelled) - Retail                        21.832122
108       Lentils - Retail                                      6.651817
94        Meat (chicken) - Retail                               9.683471
183       Meat (mutton) - Retail                               45.557209
433       Meat (veal) - Retail                                 41.270904
463       Milk (pasteurized) - Retail                           3.679687
442       Milk (powder, infant formula) - Retail              108.287696
239       Oil (olive) - Retail                                 27.274547
184       Oil (sunflower) - Retail                              8.755174
173       Onions - Retail                                       2.141539
360       Oranges - Retail                                      3.124280
112       Pasta - Retail                                        3.496637
212       Peas (green, dry) - Retail                            2.900029
83        Potatoes - Retail                                     2.297231
52        Rice - Retail                                         7.950325
185       Salt - Retail                                         2.048213
404       Spinach - Retail                                      3.240587
97        Sugar - Retail                                        4.482942
176       Tea (green) - Retail                                  4.137000
492       Tea (herbal) - Retail                                 5.434838
281       Tea - Retail                                         26.224393
317       Tomatoes (paste) - Retail                             8.085128
114       Tomatoes - Retail                                     3.393658
304       Transport (public) - Retail                           2.468031
465       Wage (non-qualified labour, non-agricultural) -...   40.548182
493       Water - Retail                                        5.384253
58        Wheat flour - Retail                                  3.253174
364       Yogurt - Retail                                       5.364898
450       Zucchini - Retail                                     3.837423

In [37]:
average_price_sorted_by_product_name = average_price.sort_values('ProductName')

# * Đề xuất:
### - Có thể phân tích mức giá trung bình của từng sản phẩm theo từng năm
### - Có thể phân tích được nơi mà các sản phẩm được sản xuất
### - Có thể sắp xếp mức giá của các món hàng từ thấp đến cao hoặc ngược lại
### - Có thể biết được giá của các mặt hàng có tăng lên hay giảm đi qua từng năm hay không